File: snippets/java-app/src/main/java/com/example/snippets/java/annotations/SnippetItem.java
package com.example.snippets.java.annotations;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Marks a method inside a [SnippetGroup] with execution metadata.
 */
@Target(ElementType.METHOD)
@Retention(RetentionPolicy.RUNTIME)
public @interface SnippetItem {
    /**
     * @return The displayed title for this individual snippet action.
     */
    String title();

    /**
     * @return A brief explanation of what the snippet achieves.
     */
    String description() default "";
}

File: snippets/kotlin-app/src/main/java/com/example/snippets/kotlin/annotations/SnippetItem.kt
package com.example.snippets.kotlin.annotations

/**
 * Marks a method inside a [SnippetGroup] with execution metadata.
 *
 * @property title The displayed title for this individual snippet action.
 * @property description A brief explanation of what the snippet achieves.
 */
@Target(AnnotationTarget.FUNCTION)
@Retention(AnnotationRetention.RUNTIME)
annotation class SnippetItem(
    val title: String,
    val description: String = ""
)
